[Brigitte Santoso] iP

brigittesantoso opened the pull request:
## DukeBot
> "Your Go-To Memory"

DukeBot helps you to keep track of things that you can't remember. It is
* text-based
* easy to learn
* **SUPER FAST** to use

Abilities:
1. Add Tasks
2. Delete Tasks
3. Mark/Unmark Tasks
4. Filter Tasks
5. And **MANY MORE**! 👍 

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you wish to practise Java coding, here's the main method:

```ruby
public static void main(String[] args) {
        new Duke("data/tasks.txt").run();
}
```
What are you waiting for? Download DukeBot [here](https://github.com/brigittesantoso/ip)! 💯

Commit:
Add Gradle support

brigittesantoso committed:
Level-1: Greet, Echo, Exit

brigittesantoso committed:
Level-2: Add, List

brigittesantoso committed:
Level-3: Mark as Done

brigittesantoso committed:
Level-4: ToDo, Event, Deadline

brigittesantoso committed:
A-TextUiTesting: Automated Text UI Testing

brigittesantoso committed:
Level-5: Handle Errors

brigittesantoso committed:
Level-6: Delete

brigittesantoso committed:
Level-5: Update Handle Errors

brigittesantoso committed:
Level-7: Save

brigittesantoso committed:
Level-8: Dates and Times

brigittesantoso committed:
Merge branch 'branch-Level-7'

brigittesantoso committed:
Merge branch 'branch-Level-8'

brigittesantoso committed:
A-MoreOOP: Use More OOP

brigittesantoso committed:
A-Packages: Organize into Packages

brigittesantoso committed:
A-JUnit: Add JUnit Tests

brigittesantoso committed:
A-JavaDoc: JavaDoc

brigittesantoso committed:
Level-9: Find

brigittesantoso committed:
Minor Changes

brigittesantoso committed:
Merge branch 'branch-Level-9'

brigittesantoso committed:
A-CodingStandard: Follow the Coding Standard

brigittesantoso committed:
Changing variables from protected to private

brigittesantoso committed:
Merge branch 'branch-A-CodingStandard'

brigittesantoso committed:
Add JavaDoc comments for tests

brigittesantoso committed:
Merge branch 'branch-A-JavaDoc'

brigittesantoso committed:
Minor Changes to Storage class

brigittesantoso committed:
A-Jar: Create a JAR File

brigittesantoso committed:
Minor Changes

brigittesantoso committed:
Merge remote-tracking branch 'origin/add-gradle-support'

brigittesantoso committed:
Set up JavaFX

brigittesantoso committed:
Level-10: GUI

brigittesantoso committed:
Merge branch 'branch-Level-10'

brigittesantoso committed:
Move Ui, Storage and Parser into package duke.main

brigittesantoso committed:
Rebuild the scene using FXML

likeabowx commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,32 @@
+/**
+ * Represents a command
+ */
+public class Command {
```
Can consider making this class an abstract class to be more OOP since it is not used as an object

likeabowx commented on src/main/java/duke/main/Duke.java:
```diff
@@ -0,0 +1,57 @@
+public class Duke {
+
+
+    private VBox dialogContainer;
```
Are these unnecessary blank lines?

likeabowx commented on src/main/java/duke/main/MainWindow.java:
```diff
@@ -0,0 +1,55 @@
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+        dialogContainer.getChildren().addAll(
+            DialogBox.getDukeDialog("Hello! I'm Duke\nWhat can I do for you?"
+                    , dukeImage));
```
The coding standard states to line break after the comma, maybe a line break isn't necessary in this case if it doesn't exceed the max characters

likeabowx commented on src/main/java/duke/main/Duke.java:
```diff
@@ -0,0 +1,57 @@
+            ui.showLoadingError(e.getMessage());
+            dialogContainer.getChildren().addAll(
+                    DialogBox.getDukeDialog(ui.toString(), duke)
+            );
```
Is egyptian style brackets necessary for this line? Maybe just a line break before the ".addAll" as stated in the guide, to break before an operator.

<img width="972" alt="image" src="https://user-images.githubusercontent.com/68813082/151763398-4886ea02-0be1-4031-b303-a018e00da467.png">

likeabowx commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,87 @@
+            String remainingCommand = fullCommand.split(" ", 2)[1];
+            return new FindCommand(remainingCommand);
+        }
+        else if(fullCommand.startsWith("mark")) {
```
The } and else if should be on the same line, applies to a few occurrences below this as well

likeabowx commented on src/main/java/duke/main/Storage.java:
```diff
@@ -0,0 +1,86 @@
+                    Task task;
+                    String line = sc.nextLine();
+                    String[] splitLine  = line.split(" | ");
+                    if (splitLine[0].equals("T")) {
+                       task = new Todo(splitLine[2]);
+                    }
+                    else if (splitLine[0].equals("D")) {
+                        task = new Deadline(splitLine[2], splitLine[3]);
+                    }
+                    else if (splitLine[0].equals("E")) {
+                        task = new Event(splitLine[2], splitLine[3]);
+                    }
+                    else {
```
Stop breaking a new line sis

likeabowx reviewed (commented):
Looks good overall, just some minor coding standard issue :blush:

brigittesantoso committed:
A-Assertions: Use Assertions

Use assert feature to document important assumptions that should hold at various points in the code.

brigittesantoso committed:
Merge pull request #1 from brigittesantoso/branch-A-Assertions

A-Assertions: Use Assertions

brigittesantoso committed:
A-CodeQuality: Improve Code Quality

Critically examines the code and refactor to improve the code quality where necessary.
Closely follows the 'Code Quality' topics learnt so far.

brigittesantoso committed:
Merge branch 'branch-A-CodeQuality'

brigittesantoso committed:
Create Update Command to easily edit tasks

brigittesantoso committed:
Revert "Create Update Command to easily edit tasks"

This reverts commit cafa490351a99614b7433d195ffe2a897ba62417.

brigittesantoso committed:
Revert "Revert "Create Update Command to easily edit tasks""

This reverts commit ea879b733e9a50bbc272ccb81979ecf694397ec4.

brigittesantoso committed:
A-UserGuide: User Guide

brigittesantoso committed:
Add Ui screenshot

brigittesantoso committed:
Minor Changes

brigittesantoso committed:
Update build.gradle and Storage class

brigittesantoso committed:
Update Exit Command

brigittesantoso committed:
Update build.gradle
